pkg/fleet/installer/packages/datadog_agent.go

```diff
@@ -287,3 +299,39 @@ func PromoteAgentExperiment(ctx context.Context) error {
 	ctx = context.WithoutCancel(ctx)
 	return StopAgentExperiment(ctx)
 }
+
+func installerCopy(path string) error {
+	// Copy the current executable to the installer path
+	// This is temporary and will be removed after next release
+	currentExecutable, err := os.Executable()
+	if err != nil {
+		return fmt.Errorf("failed to get current executable: %w", err)
+	}
+
+	sourceFile, err := os.Open(currentExecutable)
+	if err != nil {
+		return fmt.Errorf("failed to open current executable: %w", err)
+	}
+	defer sourceFile.Close()
+
+	err = os.MkdirAll(filepath.Dir(path), 0755)
+	if err != nil {
+		return fmt.Errorf("failed to create installer directory: %w", err)
+	}
+	destinationFile, err := os.Create(path)
+	if err != nil {
+		return fmt.Errorf("failed to create destination file: %w", err)
+	}
+	defer destinationFile.Close()
+
+	_, err = io.Copy(destinationFile, sourceFile)
+	if err != nil {
+		return fmt.Errorf("failed to copy executable: %w", err)
+	}
+
+	err = destinationFile.Chmod(0750)
+	if err != nil {
+		return fmt.Errorf("failed to set permissions on destination file: %w", err)
+	}
```

Comment:
Curious why you're not simply using `os.WriteFile` 🤔

Reply:
To avoid having the 50MB installer in memory 😄